daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,599 @@
+		}
+	}
+
+	// add allow incoming rules
+	for _, rule := range f.allowIncomingRules {
+		if err := f.addIncomingRule(rule); err != nil {
+			return fmt.Errorf("adding incoming rules: %w", err)
+		}
+	}
+
+	// add allow fileshare rules
+	for _, rule := range f.fileshareRules {
+		if err := f.commandExecutor.ExecuteCommand("-I " + rule); err != nil {
+			return fmt.Errorf("adding fileshare allow rule: %w", err)
+		}
+	}
+
+	f.enabled = true
+
+	return nil
+}
+
+// FileshareAllow adds ACCEPT rule for all incoming connections to tcp port 49111 from the peer with given UniqueAddress.
+func (f *FirewallManager) FileshareAllow(peer meshnet.UniqueAddress) error {
+	if _, ok := f.fileshareRules[peer.UID]; ok {
+		return fmt.Errorf("fileshare allow rule for the given peer already exists")
+	}
+
+	rule := fmt.Sprintf("INPUT -s %s/32 -p tcp -m tcp --dport 49111 -m comment --comment nordvpn -j ACCEPT", peer.Address.String())
+	if f.enabled {
+		if err := f.commandExecutor.ExecuteCommand("-I " + rule); err != nil {
+			return fmt.Errorf("adding fileshare allow rule: %w", err)
+		}
+	}
+
+	f.fileshareRules[peer.UID] = rule
+	return nil
+}
+
+// FileshareDeny removes ACCEPT rule for all incoming connections to tcp port 49111 from the peer with given UniqueAddress.
+func (f *FirewallManager) FileshareDeny(peer meshnet.UniqueAddress) error {
+	rule, ok := f.fileshareRules[peer.UID]
+	if !ok {
+		return fmt.Errorf("fileshare allow rule for the given peer doesn't exist")
+	}
+
+	if f.enabled {
+		if err := f.commandExecutor.ExecuteCommand("-D " + rule); err != nil {
+			return fmt.Errorf("removing fileshare allow rule: %w", err)
+		}
+	}
+
+	delete(f.fileshareRules, peer.UID)
+	return nil
+}
+
+func (f *FirewallManager) addBlockTrafficRules() error {
+	interfaces, err := f.devices()
+	if err != nil {
+		return fmt.Errorf("listing interfaces: %w", err)
+	}
+
+	// -I INPUT -i <iface> -m comment --comment nordvpn -j DROP
+	// -I OUTPUT -o <iface> -m comment --comment nordvpn -j DROP
+	for _, iface := range interfaces {
+		inputCommand := fmt.Sprintf("-I INPUT -i %s -m comment --comment nordvpn -j DROP", iface.Name)
+		if err := f.commandExecutor.ExecuteCommand(inputCommand); err != nil {
+			return fmt.Errorf("blocking input traffic: %w", err)
+		}
+
+		outputCommand := fmt.Sprintf("-I OUTPUT -o %s -m comment --comment nordvpn -j DROP", iface.Name)
+		if err := f.commandExecutor.ExecuteCommand(outputCommand); err != nil {
+			return fmt.Errorf("blocking output traffic: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (f *FirewallManager) BlockTraffic() error {
+	if f.trafficBlocked {
+		return fmt.Errorf("traffic is already blocked")
+	}
+
+	if f.enabled {
+		if err := f.addBlockTrafficRules(); err != nil {
+			return fmt.Errorf("adding block traffic rules: %w", err)
+		}
+	}
+
+	f.trafficBlocked = true
+
+	return nil
+}
+
+func (f *FirewallManager) removeBlockTrafficRules() error {
+	interfaces, err := f.devices()
```

Comment:
Same here, if the list of devices is different from the one in enable it will result in an error, because the rule doesn't exist and it will not remove the rules from the remaining ones. 

Wouldn't be better to have a map, like for peers, with the interfaces and the rules?